pykechain/client.py

```diff
@@ -75,12 +77,20 @@ def __init__(self, url='http://localhost:8000/', check_certificates=None):
         if check_certificates is False:
             self.session.verify = False
 
+        # Retry implementation
+        adapter = HTTPAdapter(max_retries=Retry(connect=3, backoff_factor=0.5))
+        self.session.mount('https://', adapter=adapter)
+        self.session.mount('http://', adapter=adapter)
+
     def __del__(self):
         """Destroy the client object."""
-        self.session.close()
-        del self.auth
-        del self.headers
-        del self.session
+        if hasattr(self, 'session'):
+            self.session.close()
+            del self.session
+        if hasattr(self, 'auth'):
+            del self.auth
+        if hasattr(self, 'headers'):
+            del self.headers
```

Comment:
I would ensure the existence of instance attributes by specifying them as class attributes.